client/src/components/NewMemeForm.js

```diff
@@ -32,38 +79,19 @@ const NewMemeForm = (props) => {
           const error = new Error(errorMessage);
           throw error;
         }
-      } else {
-        const body = await response.json();
-        setShouldRedirect({
-          status: true,
-          id: body.meme.id,
-        });
       }
+      const body = await response.json();
+      setNewMeme({ ...newMeme, image: body.meme });
+      setUploadedImage({ preview: "" });
```

Comment:
if you are going to redirect anyways, then you shouldnt need to update any other state! remove these lines 84 and 85